File: Samples/azure-storage/Azure.Ruby/azure_storage/models/storage_account_check_name_availability_parameters.rb
# encoding: utf-8

module Petstore
  module Models
    #
    # Model object.
    #
    class StorageAccountCheckNameAvailabilityParameters

      include MsRestAzure

      # @return [String]
      attr_accessor :name

      # @return [String]  Default value: 'Microsoft.Storage/storageAccounts' .
      attr_accessor :type


      #
      # Mapper for StorageAccountCheckNameAvailabilityParameters class as Ruby
      # Hash.
      # This will be used for serialization/deserialization.
      #
      def self.mapper()
        {
          required: false,
          serialized_name: 'StorageAccountCheckNameAvailabilityParameters',
          type: {
            name: 'Composite',
            class_name: 'StorageAccountCheckNameAvailabilityParameters',
            model_properties: {
              name: {
                required: true,
                serialized_name: 'name',
                type: {
                  name: 'String'
                }
              },
              type: {
                required: false,
                serialized_name: 'type',
                default_value: 'Microsoft.Storage/storageAccounts',
                type: {
                  name: 'String'
                }
              }
            }
          }
        }
      end
    end
  end
end

File: Samples/azure-storage/Azure.Python/storagemanagementclient/models/storage_account_check_name_availability_parameters.py
# coding=utf-8
# --------------------------------------------------------------------------
# --------------------------------------------------------------------------

from msrest.serialization import Model


class StorageAccountCheckNameAvailabilityParameters(Model):
    """StorageAccountCheckNameAvailabilityParameters.

    :param name:
    :type name: str
    :param type:  Default value: "Microsoft.Storage/storageAccounts" .
    :type type: str
    """ 

    _validation = {
        'name': {'required': True},
    }

    _attribute_map = {
        'name': {'key': 'name', 'type': 'str'},
        'type': {'key': 'type', 'type': 'str'},
    }

    def __init__(self, name, type="Microsoft.Storage/storageAccounts"):
        self.name = name
        self.type = type
